### PR TITLE
Update GridLine.py

### PR DESCRIPTION
**Pull request by lbartoletti**
Add an overlay option

### CODE REVIEW

File: python/plugins/processing/algs/qgis/GridLine.py [hunk 1]
```diff
@@ -102,35 +112,84 @@ def processAlgorithm(self, progress):
...
+            _rows = rows
+            start = 1
+        for row in xrange(start, _rows + 1):
             polyline = []
-            y = originY - (row * vSpacing)
-            for col in range(columns + 1):
-                x = originX + (col * hSpacing)
+            x = originX
+                
+            for col in xrange(start, _cols+1):
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/GridLine.py]: I think Python 3 doesn't know xrange (and python 3 is the target of master)

Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/GridLine.py]: @lbartoletti did you test this on QGIS master?

**Comment by luzpaz** [comment]
any news on when this gets merged ?

File: python/plugins/processing/algs/qgis/GridLine.py [hunk 2]
```diff
@@ -78,11 +86,15 @@ def processAlgorithm(self, progress):
...
-        originY = bbox.yMaximum()
+        originY = bbox.yMinimum()
 
         if hSpacing <= 0 or vSpacing <= 0:
             raise GeoAlgorithmExecutionException(
                 self.tr('Invalid grid spacing: %s/%s' % (hSpacing, vSpacing)))
+                
+        if hSpacing <= hOverlay or vSpacing <= vOverlay:
+            raise GeoAlgorithmExecutionException(
+                self.tr('Invalid overaly: %s/%s' % (hOverlay, vOverlay)))
```
Comment (nyalldawson) [inline python/plugins/processing/algs/qgis/GridLine.py]: "overaly" -> "overlay"

**Comment by nyalldawson** [comment]
@luzpaz Could you add a unit test covering this? See https://github.com/qgis/QGIS/blob/master/python/plugins/processing/tests/README.md

**Comment by luzpaz** [comment]
out of my scope, perhaps @lbartoletti  can do it ?

**Comment by nyalldawson** [comment]
@lbartoletti  can you please rebase this against master so that we can merge? (also #3546 too)

**Comment by nyalldawson** [comment]
@lbartoletti thanks for the updates. This PR has merge conflicts which prevent it being merged - can you please fix these?

**Comment by lbartoletti** [comment]
Yes, I will be more reactive for this...
However, is it really necessary to add "coord" field? I think it's redundant with left/right, top/bottom fields, no?

**Comment by lbartoletti** [comment]
Closed, look at #3934

**Pull request closed without merging**